### PR TITLE
SISRP-36642 - Mitigate remaining dependencies on legacy (ODS) database views

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-36642

- Removes Legacy ODS database connection. Applies logger and exception to all related methods
- Removes 'legacy_caldap' feature flag
- Leaves H2 database initialization for `test` environment due to extensive Rspec test dependencies

This doesn't remove all dependent code, that would take a lot of time. We'll have to perform that as a cleanup task at some point in the future.

### CODE REVIEW

**Comment by redconfetti** [comment]
@raydavis - The update I made to `config/initializers/populate_sisedo_h2.rb` and Canvas related Rspecs may be of interest to you. This was only necessary because I removed the CampusOracle::Queries method calls from `app/models/user/basic_attributes.rb`, but still the specs rely on the CampusOracle H2.

**Comment by raydavis** [comment]
Thank you for the ping, @redconfetti !

File: app/models/campus_oracle/queries.rb [hunk 1]
```diff
@@ -375,6 +417,8 @@ def self.terms
     private
 
     def self.get_course_sections(term_yr, term_cd, department, catalog_id, only_secondary_sections)
+      self.raise_legacy_error if Rails.env != 'test'
+
```
Comment (developish) [inline app/models/campus_oracle/queries.rb]: Do you think it's worth it to move the environment check for test into CampusOracle::Connection.raise_legacy_error itself?

```ruby
def self.raise_legacy_error
  return if Rails.env.test?
	error_msg = "CampusOracle Query call detected"
	Rails.logger.error error_msg
	raise StandardError, error_msg
end

Comment (redconfetti) [reply app/models/campus_oracle/queries.rb]: @developish - Good point

**Comment by developish** [comment]
Looks good!

**Pull request merged**